resolver/src/p2p/peer/fetcher.rs

```diff
@@ -87,12 +88,11 @@ impl<E: Clock + GClock + Rng, C: Scheme, Key: Array, NetS: Sender<PublicKey = C:
     ///
     /// If `is_new` is true, the fetch is treated as a new request.
     /// If false, the fetch is treated as a retry.
+    ///
+    /// Panics if the key is already being fetched.
     pub async fn fetch(&mut self, sender: &mut NetS, key: Key, is_new: bool) {
-        // Check if the fetch is already in progress
-        if self.active.contains_right(&key) || self.pending.contains(&key) {
-            warn!(?key, "duplicate fetch");
-            return;
-        }
+        // Panic if the key is already being fetched
+        assert!(!self.contains(&key));
```

Comment:
I think we should allow the caller to make duplicate `fetch` calls (and just drop what we already have like outstanding like before)

Reply:
Sorry for the confusion, I wanted the metric to reflect whether we were dropping the duplicate request so I had to move that check into the actor itself. So here I put assert because it would be a bug if it ever reached this code and the key already existed. Other options are:
- not doing this check
- doing this check but not panicing
But I though this was the best for now. I can add a comment explaining that it should have already been caught

Reply:
Ah got it, lost track of where this was in the call stack.